libyul/Object.cpp

```diff
@@ -38,6 +38,13 @@ using namespace solidity::langutil;
 using namespace solidity::util;
 using namespace solidity::yul;
 
+
+Object const* Object::at(YulString _name) const
+{
+	size_t subIndex = this->subIndexByName.at(_name);
```

Comment:
This can throw.

Reply:
Isn't this the expected behavior of `at` operator?

Reply:
Yeah. Is it handled properly if it throws?

Reply:
Yes, or at least, I thought so. But looking at the code again, all exceptions in yulopti are handled in the main try-catch loop as expected. However, as this code is now also shared by `Object::pathToSubObject` through the common `Object::visitPath` this may indeed throw in [EVMDialect](https://github.com/ethereum/solidity/blob/1b5c6f66930509d45c7b97ccd803e649c54a8702/libyul/backends/evm/EVMDialect.cpp#L222) when creating builtins. So yeah, good point. I will check again how the `pathToSubObject` used to work before this change to ensure the same behavior.

Reply:
It appears that the function would return an empty path if there is no [subObjectPathComponents](https://github.com/ethereum/solidity/blob/develop/libyul/Object.cpp#L142). Also, there is an assertion `object->subIndexByName.count(subObjectName)` right before the invocation of [object->at(subObjectname)](https://github.com/ethereum/solidity/pull/12941/files#diff-85786c576d9ffa5957420e25761d465d531c5e9648f3a1d3f5cb91c40e201c5eR153). Therefore, in this scenario, it should not throw an error in the `Object::at` function, instead, it should trigger the assertion failure in `visitPath`.

Reply:
But I just found a bug now. If you don't provide an object with the `--object` command-line flag, it should use the top-level object as default but it doesn't.

Reply:
Fixed. You can test with:
```
# raw Yul as input:
./build/test/tools/yulopti test/libyul/yulOptimizerTests/fullSuite/medium.yul
# Yul objects as input:
./build/test/tools/yulopti test/libyul/objectCompiler/subObjectAccess.yul
./build/test/tools/yulopti --object "A.B" test/libyul/objectCompiler/subObjectAccess.yul
./build/test/tools/yulopti --object "A.B.C" test/libyul/objectCompiler/subObjectAccess.yul
```